packages/easy-coding-standard-phar/composer.json

```diff
@@ -0,0 +1,12 @@
+{
+    "name": "symplify/easy-coding-standard-phar",
+    "description": "Use Coding Standard with 0-knowledge of PHP-CS-Fixer and PHP_CodeSniffer.",
+    "license": "MIT",
+    "require": {
+        "php": "^7.2",
+        "symplify/easy-coding-standard": "^7.3"
```

Comment:
This is not correct, since the local version is used, not after split one

Reply:
I didn't think it was going to be simple ;)

Am I correct in the assumption that the current compile pulls in the after splits of these packages?
```
"symplify/coding-standard": "^7.3",
"symplify/package-builder": "^7.3",
"symplify/set-config-resolver": "^7.3",
"symplify/smart-file-system": "^7.3",
"symplify/autowire-array-parameter": "^7.3",
"symplify/auto-bind-parameter": "^7.3"
```
so the phar compile should/can pull in the aftersplit of "symplify/easy-coding-standard": "^7.3"

Locally the phar can be compiled and runs correctly

An alternative may be modifying the composer.json of easy-coding-standard when compiling the phar file , adding the replace block only when building the phar that would require a lot less changes I guess

Reply:
> Am I correct in the assumption that the current compile pulls in the after splits of these packages?

This is probably right, while it should use current `master`.

> so the phar compile should/can pull in the aftersplit of "symplify/easy-coding-standard": "^7.3"

That would be wrong. Imagine you'd fix a bug in ECS:

- the PR gets merged
- then the `ecs.phar` compile begins
- then it downloads old version of ECS that still has the bug
- the bugged `ecs.phar` is releated
- then the split happens (race condition)
- then fixed ECS is published

Reply:
true, so the building of the phar should be run after the publication, but that would be true for the other libraries aswell right? They have to be published too before the PHAR gets built the scenario is the same I guess, in both cases the packages have to be published before the PHAR gets built right? I'm not familliar enough with the build process yet to make a judgement call here.

Or we should go for the easier (slightly more hacky) solution which is modify composer.json on the fly at compile time adding the replace block only for building the PHAR.

Reply:
As this is monorepo, the phar should be build from monorepo code. The publication is completely unrelated

Reply:
> As this is monorepo, the phar should be build from monorepo code. The publication is completely unrelated

I completely understand, but as it stands in the current situation, the current PHAR isn't completely built from the monorepo either I think, It's built from the easy-coding-standard monorepo code, but when it's built it pulls in the published versions of it's dependent Simplify libraries, which are also in the monorepo. If it should only use monorepo code than it should copy those dependencies into the PHAR when it is being built right?
Copying instead of requiring could in fact also be used to solve the same problem. When building the PHAR, instead of requiring the Symplify libraries, we copy them from the monorepo, tell composer where to find them and make sure that all dependencies are defined in the composer.json file for the phar including the replace block. That way the PHAR is always built from the current monorepo code and only pulls in external dependencies.

Reply:
This might help https://www.tomasvotruba.com/blog/2020/02/10/how-to-test-monorepo-after-split-before-actual-split/

Reply:
will have a look later this evening, thanks!